From e731039f71fdb4927b488cf88e61c81b8355446b Mon Sep 17 00:00:00 2001
From: SamarV-121 <samarvispute121@pm.me>
Date: Tue, 16 Aug 2022 10:07:33 +0200
Subject: [PATCH] StorageManager: Add dummy inCryptKeeperBounce method

* Fixes mtk ims crash
E AndroidRuntime: java.lang.NoSuchMethodError: No static method inCryptKeeperBounce()Z in class Landroid/os/storage/StorageManager; or its super classes (declaration of 'android.os.storage.StorageManager' appears in /system/framework/framework.jar!classes2.dex)

Change-Id: Ia923c10b79c16a41fe4e8dd454e96afa05e174e2
---
 core/java/android/os/storage/StorageManager.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/java/android/os/storage/StorageManager.java b/core/java/android/os/storage/StorageManager.java
index 497bfa6380bc..a21e2ba9ee4c 100644
--- a/core/java/android/os/storage/StorageManager.java
+++ b/core/java/android/os/storage/StorageManager.java
@@ -1693,6 +1693,10 @@ public class StorageManager {
         return RoSystemProperties.CRYPTO_ENCRYPTED;
     }
 
+    public static boolean inCryptKeeperBounce() {
+        return false;
+    }
+
     /** {@hide}
      * Is this device file encrypted?
      * @return true for file encrypted. (Implies isEncrypted() == true)
-- 
2.37.2

